src/deepsparse/open_pif_paf/utils/annotate.py

```diff
@@ -0,0 +1,193 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    import openpifpaf
+
+    openpifpaf_error = None
+except ModuleNotFoundError as openpifpaf_import_error:
+    openpifpaf = None
+    openpifpaf_error = openpifpaf_import_error
+
+import logging
+from typing import List, Optional, Tuple
+
+import numpy
+
+import cv2
+import torch
+from deepsparse.open_pif_paf.schemas import OpenPifPafOutput
+from deepsparse.yolact.utils.annotate import (
+    _get_text_size,
+    _plot_fps,
+    _put_annotation_text,
+)
+from openpifpaf import decoder, network
+
+
+_LOGGER = logging.getLogger(__name__)
+
+# necessary openpifpaf dependencies for now
+model_cpu, _ = network.Factory().factory(head_metas=None)
+processor = decoder.factory(model_cpu.head_metas)
+
+
+def _get_color():
+    return tuple(numpy.random.randint(0, 256, 3).tolist())
+
+
+# assign a color to a group of body parts
+BODY_PART_COLOR = {
+    "ankle_knee": _get_color(),
+    "knee_hip": _get_color(),
+    "shoulder_elbow": _get_color(),
+    "elbow_wrist": _get_color(),
+    "nose_eye:": _get_color(),
+    "eye_ear": _get_color(),
+    "ear_shoulder": _get_color(),
+    "other": _get_color(),
+}
+
+CONFIDENCE_BOX_COLOR = _get_color()
+
+
+def annotate_image(
+    image: numpy.ndarray,
+    prediction: OpenPifPafOutput,
+    images_per_sec: Optional[float] = None,
+    **kwargs,
+) -> numpy.ndarray:
+
+    cafs, cifs = prediction.caf, prediction.cif
+    # we expect to have received batch = 1 from the pipeline
+    if not (cafs.shape[0] == 1 and cifs.shape[0] == 1):
+        raise ValueError("Expected batch size of 1, got {}".format(cafs.shape[0]))
+    annotations = processor._mappable_annotations(
+        [torch.tensor(cifs[0]), torch.tensor(cafs[0])], None, None
+    )
+
+    input_image_resolution = image.shape[:2]
+    scale = numpy.flipud(
+        numpy.divide(input_image_resolution, kwargs["model_resolution"])
+    )
+
+    for annotation in annotations:
+        data = numpy.multiply(annotation.data[:, :2], scale)
+        keypoints = annotation.keypoints
+        image = _draw_skelethon(image, annotation.skeleton, data, keypoints)
+        image = _draw_joints(image, data)
+        image = _draw_confidence(image, data[0], annotation.score)
+        if images_per_sec:
+            image = _plot_fps(
+                img_res=image,
+                images_per_sec=images_per_sec,
+                x=20,
+                y=30,
+                font_scale=0.9,
+                thickness=2,
+            )
+    return image
+
+
+def _draw_confidence(
```

Comment:
are all these private functions modified from openpifpaf repo? if so optional nit is to say so in docstrings

Reply:
No, I wrote them. Note how some of them reuse our previous functions :)

Reply:
Oh awesome! 👍